shared/src/simulation/activities/provide-personnel-from-vehicles.ts

```diff
@@ -192,7 +190,7 @@ function personnelInUnloadingVehicles(
                 tryGetElement(draftState, 'personnel', personnelId)
                     ?.personnelType
         )
-        .filter((pt): pt is PersonnelType => pt !== undefined)
-        .forEach((pt) => resource[pt]++);
+        .filter((pt): pt is string => pt !== undefined)
```

Comment:
This type guard is unnecessary

```suggestion
        .filter((pt) => pt !== undefined)
```